src/app/pages/RoflAppDetailsPage/Endorsement.tsx

```diff
@@ -30,77 +63,111 @@ export const Endorsement: FC<EndorsementProps> = ({ endorsements }) => {
+        return (
+          <TextBox>
+            <Trans t={t} i18nKey={`rofl.endorsementLabels.${key}`} />
+          </TextBox>
+        )
       case 'provider_instance_admin':
-        return t('rofl.endorsementLabels.provider_instance_admin')
-      case 'and':
-        return t('rofl.endorsementLabels.and')
-      case 'or':
-        return t('rofl.endorsementLabels.or')
-    }
-    exhaustedTypeWarning('Unknown endorsement', key)
-    return key
-  }
-
-  const getValue = (key: RoflAllowedEndorsementFields, value: any) => {
-    switch (key) {
-      case 'any':
-        return ''
-      case 'role_compute':
-        return ''
-      case 'role_observer':
-        return ''
-      case 'entity':
-        return <Typography variant="mono">{value}</Typography>
-      case 'node':
-        return <Typography variant="mono">{value}</Typography>
       case 'provider':
-        return <Typography variant="mono">{value}</Typography>
-      case 'provider_instance_admin':
-        return <Typography variant="mono">{value}</Typography>
-      case 'and':
+      case 'node':
         return (
-          <Box sx={{ ml: 4 }}>
-            <Endorsement endorsements={value} />
-          </Box>
+          <TextBox>
+            <Trans
+              t={t}
+              i18nKey={`rofl.endorsementLabels.${key}`}
+              components={{
+                address: <AccountLink scope={scope} address={value} alwaysTrim preferAdaptive />,
+              }}
+            />
+          </TextBox>
+        )
+      case 'entity':
+        return (
+          <TextBox>
+            <Trans
+              t={t}
+              i18nKey="rofl.endorsementLabels.entity"
+              components={{
+                address: (
+                  <AccountLink
+                    scope={{ network: scope.network, layer: 'consensus' }}
+                    address={getOasisAddressFromBase64PublicKey(value)}
+                    alwaysTrim
+                    preferAdaptive
+                  />
+                ),
+              }}
+            />
+          </TextBox>
         )
+      case 'and':
       case 'or':
         return (
-          <Box sx={{ ml: 4 }}>
-            <Endorsement endorsements={value} />
+          <Box
+            component={'div'}
+            sx={{
+              flex: 1,
+              display: 'flex',
+              flexDirection: 'column',
+              overflowX: 'hidden',
+              paddingLeft: '15px',
+              position: 'relative',
+              '&::before': {
+                content: '""',
+                position: 'absolute',
+                left: 0,
+                top: 0,
+                height: '100%',
+                width: '20px' /* Width of the brace area */,
+                borderLeft: '1px solid #000' /* Vertical line of the brace */,
+                boxSizing: 'border-box',
+                borderTopLeftRadius: '10px',
+                borderBottomLeftRadius: '10px',
+              },
```

Comment:
same as?
```suggestion
              borderLeft: '1px solid #000' /* Vertical line of the brace */,
              borderTopLeftRadius: '10px',
              borderBottomLeftRadius: '10px',
```

also, `content: '""'` looks suspicious